Sources/Controllers/RoutePlanning/OAGpxApproximationViewController.mm

```diff
@@ -145,139 +132,59 @@ - (void)initData
 {
     NSMutableArray<OAApplicationMode *> *profiles = [NSMutableArray arrayWithArray:OAApplicationMode.values];
     [profiles removeObject:OAApplicationMode.DEFAULT];
-	NSMutableArray<OAApplicationMode *> *toRemove = [NSMutableArray array];
+    NSMutableArray<OAApplicationMode *> *toRemove = [NSMutableArray array];
     [profiles enumerateObjectsUsingBlock:^(OAApplicationMode *profile, NSUInteger ids, BOOL *stop) {
         if ([profile.getRoutingProfile isEqualToString:@"public_transport"])
-			[toRemove addObject:profile];
+            [toRemove addObject:profile];
     }];
-	[profiles removeObjectsInArray:toRemove];
+    [profiles removeObjectsInArray:toRemove];
     return profiles;
 }
 
-- (BOOL)setSnapToRoadAppMode:(OAApplicationMode *)appMode
+- (void)didStartProgress
```

Comment:
`OAGpxApproximationHelperDelegate` should always call methods from the main thread, so you do not need to care about `dispatch_async(dispatch_get_main_queue()` inside implementation
Recheck and fix if needed